lesson_7/szilard_vig/decorators.py

```diff
@@ -0,0 +1,71 @@
+def join_args(args):
+    return ','.join(repr(arg) for arg in args)
+
+#Create a decorator called uppercase that will uppercase the result
+def uppercase(fnc):
+    fnc_name = fnc.__name__
+    print('Decorating function {}'.format(fnc_name))
+    def inner(*args):
+        print('Decorated function {} got the following args: {}'.format(fnc_name, join_args(args)))
+        text = fnc(*args)
+        return text.upper()
+    return inner
+
+@uppercase
+def greet1(name):
+    return "Greetings {}!".format(name)
+
+print(greet1('Earhling'))
+
+print('-'*20)
+#Create a decorator called safe_divide that will output a message if the division cannot be per-formed, othervise it will return the result.
+def safe_divide(fnc):
+    fnc_name = fnc.__name__
+    print('Decorating function {}'.format(fnc_name))
+    def inner(first_number, second_number):
+        try:
+            return first_number / second_number
+        except:
```

Comment:
You can specify which exception you want to catch. Catching all exception can lead to bugs 😄 

Try to run the code without the `try except` and see what exception you get and use that in the `except` statement